File: Leetcode/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minSoFar = prices[0];
        int profit = 0;

        for (int i = 1; i <  prices.size(); i++) {
            minSoFar = min(minSoFar, prices[i]);
                         
            profit = max(profit, prices[i] - minSoFar);
        }
        return profit;
    }
};

File: Leetcode/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = prices[0];
        int profit = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i] < minPrice){
                minPrice = prices[i];
            }
            if(prices[i]-minPrice > profit)
            profit = prices[i]-minPrice;
        }
        return profit;
    }
}
